scene/animation/animation_tree.h

```diff
@@ -114,9 +113,10 @@ class AnimationNode : public Resource {
 	GDVIRTUAL1RC(bool, _is_parameter_read_only, StringName)
 	GDVIRTUAL4RC(double, _process, double, bool, bool, bool)
 	GDVIRTUAL0RC(String, _get_caption)
-	GDVIRTUAL0RC(bool, _has_filter)
 
 public:
+	StringName filter = PNAME("filter");
```

Comment:
I don't like having a member variable named the same as its string constant value. If this is effectively a constant, it should be marked `const` and made capital, or renamed to `_string` or `_parameter` to indicate that it refers to the parameter / string constant, and it isn't itself a filter object or a time value.

The existing code is also bad practice and should be fixed in a separate PR. In particular, the use of `time` as a member variable name shadows the C library's `time(time_t*)` function. I wouldn't have as much of an issue with this additional parameter as long as it is next to the other PNAME values, namely `time`, `remaining` and so on.

This issue isn't a sticking point, just something I observed that made it especially difficult for me to read and follow the property list code.

Reply:
I agree your opinion, here I just abey the existing code.